src/System.Console/tests/SyncTextReader.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class SyncTextReaderTests
+{
+    // NOTE: These tests test the underlying SyncTextReader by
+    // accessing the Console.In TextReader (which in fact is a SyncTextReader).
+
+    static readonly string[] s_testLines = new string[] {
+        "3232 Hello32 Hello 5032 Hello 50 532 Hello 50 5 aTrueaabcdbc1.23123.4561.23439505050System.ObjectHello World",
+        "32",
+        "",
+        "32 Hello",
+        "",
+        "32 Hello 50",
+        "",
+        "32 Hello 50 5",
+        "",
+        "32 Hello 50 5 a",
+        "",
+        "True",
+        "a",
+        "abcd",
+        "bc",
+        "1.23",
+        "123.456",
+        "1.234",
+        "39",
+        "50",
+        "50",
+        "50",
+        "System.Object",
+        "Hello World",
+    };
+
+    private static void Test(string content, Action<StreamReader> action)
```

Comment:
Do any of the caller's use the StreamReader that's passed in to action?  It seems like the goal is for the caller to use Console.In rather than the underlying StreamReader you create, and none of your current callers are using this StreamReader.